camel-core/src/main/java/org/apache/camel/builder/ExpressionBuilder.java

```diff
@@ -1895,13 +1920,25 @@ public Object evaluate(Exchange exchange) {
                     throw new IllegalArgumentException("Command not supported for dateExpression: " + command);
                 }
 
-                SimpleDateFormat df = new SimpleDateFormat(pattern);
-                return df.format(date);
+                // Apply offsets
+                long dateAsLong = date.getTime();
+                for (long offset : offsets) {
+                    dateAsLong += offset;
+                }
+                date = new Date(dateAsLong);
+
+                if (null != pattern && !pattern.isEmpty()) {
```

Comment:
We use coding style where null is on the other side of the if, eg pattern != null